src/common/errors_js.ml

```diff
@@ -131,13 +133,89 @@ let print_reason_color ~first ~one_line ~color (message: message) =
   (if first then Printf.printf "\n");
   C.print ~color_mode:color to_print
 
-let print_error_color ~one_line ~color (e : error) =
+let print_error_color_old ~one_line ~color (e : error) =
   let {kind; messages; trace} = e in
   let messages = prepend_kind_message messages kind in
   let messages = append_trace_reasons messages trace in
   print_reason_color ~first:true ~one_line ~color (List.hd messages);
   List.iter (print_reason_color ~first:false ~one_line ~color) (List.tl messages)
 
+let file_location_style text = (C.Underline C.Default, text)
+let default_style text = (C.Normal C.Default, text)
+let source_fragment_style text = (C.Normal C.Default, text)
+let error_fragment_style text = (C.Normal C.Red, text)
+let line_number_style text = (C.Dim C.Default, text)
+
+let relative_path filename =
+  let relname = Files_js.relative_path (Sys.getcwd ()) filename in
+  if String.length relname < String.length filename
+    then relname
+    else filename
+
+let highlight_error_in_line line c0 c1 =
+  let prefix = String.sub line 0 c0 in
+  let fragment = String.sub line c0 (c1 - c0) in
+  let suffix = String.sub line c1 ((String.length line) - c1) in
+  [
+    source_fragment_style prefix;
+    error_fragment_style fragment;
+    source_fragment_style suffix;
+  ]
+
+(* TODO: Ideally this would also work if the file was read from stdin *)
```

Comment:
So what happens when you run

  flow check-contents < foo.js

? If that needs love, then I think you could actually pass along the stdin contents. Take a look at [where the code for check-contents calls into this stuff](https://github.com/facebook/flow/blob/master/src/commands/checkContentsCommand.ml#L55). You could probably pass the `file` into `print_error_summary`. Then you can compare the `file`'s name to the file you're trying to read. If `file` doesn't have a name, then it will probably show up in the error as `-`.

Here are more examples of check-contents: https://gist.github.com/gabelevi/d77c00701eb7dd3429d8